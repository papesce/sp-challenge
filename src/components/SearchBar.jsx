import React, { Component } from "react";
import TextField from 'material-ui/TextField';
import {connect} from 'react-redux'
import {search, searchClear} from '../redux/actions'
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
    bar: {
      marginLeft: 5* theme.spacing.unit,
      marginRight:  5 * theme.spacing.unit,
      marginTop: 4 *  theme.spacing.unit,
      width: "90%"
    },
})

export class SearchBar extends Component {
  constructor(props) {
    super(props)
    this.handleTextChanged = this.handleTextChanged.bind(this);
  }
  handleTextChanged(event) {
    const text = event.target.value;
    if (text.trim() === "")
      this.props.clear();  
    else {
      this.props.search(text);
    }
  }
  render() {
    const {classes} = this.props;
    return (
      <div >
        <TextField 
          className= {classes.bar}
          id="search"
          label="Enter text to search"
          type="search"
          //npm className={classes.textField}
          margin="normal"
          onChange={this.handleTextChanged}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
     search :  (text) => dispatch(search(text)),
     clear : () => dispatch(searchClear())
  }
}

export const SearchBarS = withStyles(styles)(SearchBar)

export default connect(undefined, mapDispatchToProps)(SearchBarS)