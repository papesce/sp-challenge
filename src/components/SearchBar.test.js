import React from 'react'
import { SearchBarS } from './SearchBar'
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';

describe('SearchBar tests', () => {
    it('should render', () => {  
        shallow(<SearchBarS/>);
    })
    it('renders correctly', () => {
        const tree = renderer.create(<SearchBarS/>)
            .toJSON();
         expect(tree).toMatchSnapshot();
    });
    
})