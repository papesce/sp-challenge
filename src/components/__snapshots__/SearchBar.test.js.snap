// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests renders correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root-2 MuiFormControl-marginNormal-3 SearchBar-bar-1"
    onBlur={[Function]}
    onFocus={[Function]}
    required={false}
  >
    <label
      className="MuiFormLabel-root-12 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10"
      data-shrink={false}
      htmlFor="search"
    >
      Enter text to search
    </label>
    <div
      className="MuiInput-root-16 MuiInput-formControl-17 MuiInput-inkbar-18 MuiInput-underline-24"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <input
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-20 MuiInput-inputSingleline-27 MuiInput-inputSearch-28"
        defaultValue={undefined}
        disabled={false}
        id="search"
        name={undefined}
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="search"
        value={undefined}
      />
    </div>
  </div>
</div>
`;
